one_fm/one_fm/hr_utils.py

```diff
@@ -21,38 +27,96 @@ def indemnity_allocation_builder(employee_list):
+    if total_working_year < 5 or (total_working_year == 5 and total_working_days == 5*365):
+        #15 days salary is divided over a year and  that becomes each day's allocation.
+        total_amount = 15 * per_day_amount / 365 * total_working_days 
+    
+    elif total_working_year >= 5 and total_working_days > 5*365:
+        #calculation takes 15 days salary for 5 years and 30 days salary after 5 years 
+        amount_1 = 15 * per_day_amount / 365 * five_year_in_days
+        amount_2 = 30 * per_day_amount / 365 * (total_working_days-five_year_in_days)
+        total_amount = amount_1+amount_2
+    return total_amount
+
+def get_per_day_indemnity_amount(date_of_joining, to_date, indemnity_amount):
+    """To Calculate indemnity of the employee per day distributed across one year. 
+    This allows to get the per day calculation to be allocated every day.
+
+    Args:
+        date_of_joining ([date]): Employee's Joining Date
+        to_date ([data]): up until date
+        indemnity_amount ([currency]): Indemnity Amount from Salary Structure Assignment
+
+    Returns:
+        [type]: [description]
+    """
+    total_working_year = relativedelta(to_date, date_of_joining).years
+    total_working_days = (to_date - date_of_joining).days
+    per_day_amount = indemnity_amount/26
+
+    #calculate indemnity per day.
+    if total_working_year < 5:
+        return 15 * per_day_amount / 365
+    elif total_working_year >= 5 and total_working_days > (5*365):
+        return 30 * per_day_amount / 365 
 
 def allocate_daily_indemnity():
     # Get List of Indemnity Allocation for today
     allocation_list = frappe.get_all("Indemnity Allocation", filters={"expired": ["!=", 1]}, fields=["name"])
     for alloc in allocation_list:
         allow_allocation = True
         allocation = frappe.get_doc('Indemnity Allocation', alloc.name)
-        # Check if employee absent today then not allow allocation for today
-        is_absent = frappe.db.sql("""select name, status from `tabAttendance` where employee = %s and
-            attendance_date = %s and docstatus = 1 and status = 'Absent' """,
-			(allocation.employee, getdate(nowdate())), as_dict=True)
-        if is_absent and len(is_absent) > 0:
-            allow_allocation = False
-
-        # Check if employee on leave today and allow allocation for today
-        leave_appln = frappe.db.sql("""select name, status, leave_type from `tabLeave Application` where employee = %s and
-            (%s between from_date and to_date) and docstatus = 1 and status = 'Rejected'""",
-			(allocation.employee, getdate(nowdate())), as_dict=True)
-        if leave_appln and len(leave_appln) > 0:
-            allow_allocation = False
-
-        if allow_allocation:
-            # Set Daily Allocation
-            allocation.new_indemnity_allocated = 30/365
-            allocation.total_indemnity_allocated = allocation.total_indemnity_allocated+allocation.new_indemnity_allocated
-            allocation.save()
```

Comment:
@SalShai @mymi14s Asper the requirement, Indemnity will not be allocated for absent or Leave days !
Also Here we are calculating the factor of indemnity not the amount

Reply:
@pjamsheer please check up with @SalShai as the story was accepted, if any differences, it can be hotfixed.

Reply:
@pjamsheer , According to the new requirement, Indemnity was calculated based on his no. of working years. if no. of working year is 5 or less, he gets 15 day's pay. above 5, he gets 30. They also asked us to ignore the fact that the employee was absent. They are just considering the no. of days he is working for the company.

Reply:
Okay, Great.